### ISSUE TITLE
Docs contain predicate logic example using none? that doesn't work

### ISSUE THREAD

**Issue by johanlunds** [dry-rb/dry-schema#376]
## Describe the bug

Here:

https://github.com/dry-rb/dry-schema/blob/f5c2dbdb1374193792272f99995e0a68f63ddaf3/docsite/source/advanced/predicate-logic.html.md?plain=1#L25

```ruby
Dry::Schema.Params do
  required(:age) { none? | int? }
end
```

However `none?` doesn't seem supported just by looking through the remaining docs and by looking through the code. I am guessing the name was changed to `nil?` - that's what the [tests seem to indicate](https://github.com/dry-rb/dry-schema/blob/f5c2dbdb1374193792272f99995e0a68f63ddaf3/spec/integration/schema/predicates/nil_spec.rb).

Further, using `none?` has very weird behavior. It sort of fools you it works initially because it doesn't generate any errors or warnings, but actually using it generates weird results and errors. Very weird. See example below how to reproduce.

## To Reproduce

```ruby
s = Dry::Schema.Params do
  required(:age) { none? | int? }
end

s.({})                     # => #<Dry::Schema::Result{} errors={:age=>["is missing"]} path=[]>
# 🟢 OK!
s.(age: 123)               # => #<Dry::Schema::Result{:age=>123} errors={} path=[]>
# 🟢 OK!
s.(age: '123')             # => #<Dry::Schema::Result:0x113c8>
# 🚨 WEIRD...
s.(age: '123').to_h        # => {:age=>"123"}
# 🚨 SEEMS WRONG! Should only allow none or int?
s.(age: 123).to_h          # => {:age=>123}
# 🟢 OK!
s.(age: '123').errors.to_h # => Dry::Schema::MissingMessageError: Message template for :age under "" was not found. Searched in: ...
# 🚨 SEEMS WRONG! Generates error
s.(age: nil)               # => #<Dry::Schema::Result{:age=>nil} errors={} path=[]>
# 🟢 OK!
s.(age: 23.5)              # => #<Dry::Schema::Result:0x113dc>
# 🚨 WEIRD AGAIN...
s.(age: 23.5).errors       # => Dry::Schema::MissingMessageError: Message template for :age under "" was not found. Searched in: ...
# 🚨 SEEMS WRONG! Same error again
```

## Expected behavior

I would expect two or three things:

1. Remove `none?` from docs if it's not supported
2. When the schema is defined, maybe raise an error if the schema tries to use unsupported or invalid predicate logic?
3. That the schema result would not return invalid values (like floats and strings in the example above) AND return proper error messages without throwing an error.

## My environment

- Affects my production application: YES (although change is not deployed, found during QA and code review)
- Ruby version: `ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [aarch64-linux-musl]`
- OS: `Alpine release 3.13.6`
- Dry gem versions:

They are all the latest versions:

```
dry-configurable (0.13.0)
dry-container (0.9.0)
dry-core (0.7.1)
dry-events (0.3.0)
dry-inflector (0.2.1)
dry-initializer (3.0.4)
dry-logic (1.2.0)
dry-matcher (0.9.0)
dry-monads (1.4.0)
dry-schema (1.8.0)
dry-struct (1.4.0)
dry-transaction (0.13.3)
dry-types (1.5.1)
dry-validation (1.7.0)
```

**Comment by flash-gordon** [comment]
Docs should be updated, that's true. However, you seem to have wrong assumptions about how dry-schema works. `Dry::Schema::Result#to_h` doesn't return values only if they are valid, it returns output regardless. This means you need to check if a result value is a `success?` or a `failure?` before calling `to_h`.

Compare:
```ruby
s = Dry::Schema.Params do
  required(:age) { nil? | int? }
end
s.(age: '123').to_h 
# => {:age=>"123"} 
# still invalid input even when `nil?` is used
s.(age: '123') 
# => false
```

**Comment by johanlunds** [comment]
Ah of course! I didn't stop to consider that. I agree with you then; just the docs that need updating.

**Comment by johanlunds** [comment]
Opened a pull request to update the docs @flash-gordon
